components/sync15/src/request.rs

```diff
@@ -396,7 +424,7 @@ where
             }
         };
 
-        if want_commit || self.batch == BatchState::Unsupported {
+        if want_commit{
```

Comment:
Move this into the following if statement, since now there's no condition on batch state.